include/rppdefs.h

```diff
@@ -729,6 +735,58 @@ typedef struct RpptResamplingWindow
     __m128 pCenter, pScale;
 } RpptResamplingWindow;
 
+struct BaseMelScale
+{
+    public:
+        inline RPP_HOST_DEVICE virtual Rpp32f hz_to_mel(Rpp32f hz) = 0;
+        inline RPP_HOST_DEVICE virtual Rpp32f mel_to_hz(Rpp32f mel) = 0;
+        virtual ~BaseMelScale() = default;
+};
+
+struct HtkMelScale : public BaseMelScale
+{
+    inline RPP_HOST_DEVICE Rpp32f hz_to_mel(Rpp32f hz) { return 1127.0f * std::log(1.0f + (hz / 700.0f)); }
+    inline RPP_HOST_DEVICE Rpp32f mel_to_hz(Rpp32f mel) { return 700.0f * (std::exp(mel / 1127.0f) - 1.0f); }
+    public:
+        ~HtkMelScale() {};
+};
+
+struct SlaneyMelScale : public BaseMelScale
```

Comment:
Are we sure we need these structs with all this compute code in a high level rppdefs.h header thats's used for just one functionality?
@sampath1117 Please do a round of review on this PR

Reply:
@r-abishek 
We moved this to rppdefs.h because the same code is used in both HOST and HIP

Please let us know if there is any other place where we can move this to

Reply:
Okay, for now just add the doxygen tag for this struct like in RpptResamplingWindow above

Reply:
Done